### PR TITLE
fix: hasRoute method comparison with case insensitive

### PR DESCRIPTION
**Pull request by SMNBLMRR**
fix for issue: #5503 
#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/route.js [hunk 1]
```diff
@@ -152,8 +152,9 @@ function buildRouting (options) {
   }
 
   function hasRoute ({ options }) {
+    const normalizeMethod = (options.method ?? '').toUpperCase()
```
Comment (Uzlopak) [inline lib/route.js:155]: Rename to normalizedMethod.

Comment (Uzlopak) [reply lib/route.js:155]:
```suggestion
    const normalizedMethod = options.method ? options.method.toUpperCase() : ''
```

File: lib/route.js [hunk 2]
```diff
@@ -152,8 +152,9 @@ function buildRouting (options) {
...
     return router.hasRoute(
-      options.method,
+      normalizeMethod,
```
Comment (Uzlopak) [inline lib/route.js:157]:
```suggestion
      normalizedMethod,
```

File: test/has-route.test.js [hunk 3]
```diff
@@ -74,4 +74,20 @@ test('hasRoute', t => {
       url: '/example/:file(^\\d+).png'
     }), true)
   })
+
+  test('hasRoute - with normalize method', t => {
```
Comment (Uzlopak) [inline test/has-route.test.js:78]:
```suggestion
  test('hasRoute - finds a route even if method is not uppercased', t => {
```

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/route.js [hunk 4]
```diff
@@ -152,8 +152,9 @@ function buildRouting (options) {
   }
 
   function hasRoute ({ options }) {
+    const normalizedMethod = options.method ? options.method.toUpperCase() : ''
```
Comment (jsumners) [inline lib/route.js:155]:
```suggestion
    const normalizedMethod = options.method?.toUpperCase() ?? ''
```

Comment (SMNBLMRR) [reply lib/route.js:155]: Ah yes! I agree with you. I'll update as you suggested.

**Comment by gurgunday** [comment]
`backport` label will work after the PR merges

**Review by gurgunday — APPROVED** [review]

**Review by jsumners — APPROVED** [review]

**Review by gurgunday — APPROVED** [review]
lgtm

**Comment by SMNBLMRR** [comment]
@gurgunday Should I try to do this manually?

**Comment by gurgunday** [comment]
Yeah, it would be great

I think v4 has a different findRoute implementation but it shouldn't be a problem, it just prevents the auto-backport

**Comment by SMNBLMRR** [comment]
Alright!, I have created a new PR for v4 #5513

**Pull request merged**